src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorLSPClientOptionsMonitor.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.ComponentModel.Composition;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor
+{
+    /// <summary>
+    /// Keeps track of accurate settings on the client side so we can easily retrieve the
+    /// options later when the server sends us a workspace/configuration request.
+    /// </summary>
+    [Export(typeof(RazorLSPClientOptionsMonitor))]
```

Comment:
Use the `System.Composition` export type and throw a [Shared] above this to ensure it doesn't get treated transiently.